mlir/lib/Dialect/Tosa/Transforms/TosaFolders.cpp

```diff
@@ -202,7 +191,51 @@ LogicalResult notifyIfNoTosaDenseConstantTensor(Value toCheck,
 }
 
 template <typename BaseType>
-DenseElementsAttr transposeType(ElementsAttr attr, ShapedType inputType,
+DenseElementsAttr transposeTypeRaw(DenseElementsAttr attr, ShapedType inputType,
+                                ShapedType outputType,
+                                llvm::ArrayRef<int64_t> permValues) {
+  if (inputType.getNumElements() == 0)
+    return DenseElementsAttr::get(outputType, llvm::ArrayRef<BaseType>{});
+
+  auto inputShape = inputType.getShape();
+
+  // The inverted permutation map and strides of the output are used to compute
+  // the contribution of a given dimension to the destination linear index in
+  // an order-independent way.
+  auto outputStrides = computeStrides(outputType.getShape());
+  auto invertedPermValues = invertPermutationVector(permValues);
+
+  //auto attrValues = attr.getValues<BaseType>();
+  //auto initialValue = *std::begin(attrValues);
+  SmallVector<BaseType> outputValues(inputType.getNumElements());
+
+  auto* raw = reinterpret_cast<const BaseType*>(attr.getRawData().data());
```

Comment:
Are you allowed to use https://en.cppreference.com/w/cpp/numeric/bit_cast? The `reinterpret_cast` is not strictly sensible.

Reply:
Why do you think that the reinterpret_cast is not sensible? `char` is allowed to alias with any other type.

Reply:
I thought it should be `static_cast` - thanks for the link to point 5) of https://en.cppreference.com/w/cpp/language/reinterpret_cast
> 5) Any object pointer type T1* can be converted to another object pointer type cv T2*. This is exactly equivalent to static_cast<cv T2*>(static_cast<cv void*>(expression)) (which implies that if T2's alignment requirement is not stricter than T1's, the value of the pointer does not change and conversion of the resulting pointer back to its original type yields the original value). In any case, the resulting pointer may only be dereferenced safely if allowed by the type aliasing rules (see below).